src/platform/macos/input.cpp

```diff
@@ -379,10 +383,15 @@ const KeyCodeMap kKeyCodesMap[] = {
 
   void
   abs_mouse(input_t &input, const touch_port_t &touch_port, float x, float y) {
-    auto scaling = ((macos_input_t *) input.get())->displayScaling;
+    auto macos_input = static_cast<macos_input_t *>(input.get());
+    auto scaling = macos_input->displayScaling;
+    auto display = macos_input->display;
 
     CGPoint location = CGPointMake(x * scaling, y * scaling);
-
+    CGRect display_bounds = CGDisplayBounds(display);
+    // in order to get the correct mouse location for capturing display , we need to add the display bounds to the location
```

Comment:
The location we received are relative to the display we are capturing ,so we get the display bounds using [CGDisplayBounds](https://developer.apple.com/documentation/coregraphics/1456395-cgdisplaybounds?language=objc) then add them up to get the right input coordinates.